pre-docs/writing_tasks.md

```diff
@@ -361,6 +361,71 @@ Successful on 1 node: localhost
 Ran on 1 node in 0.12 seconds
 ```
 
+### Remote Tasks
+
+Some targets are hard or impossible to execute tasks on directly. For example a
+network device may have a limited shell environment or a cloud service may be
+driven only by HTTP APIs. In these cases it makes sense to write a task that
+runs on a proxy target and remotely interacts with real target. By writing a
+remote task Bolt allows users to specify connection information for remote
+targets in their inventory file and injects them into the `_target` metaparam.
+
+The following example shows how to write a task that posts messages to slack
+and reads connection information from inventory.yaml
+
+```ruby
+#!/usr/bin/env ruby
+# modules/slack/tasks/message.rb
+
+require 'json'
+require 'net/http'
+
+params = JSON.parse(STDIN.read)
+# the slack API token is passed in from inventory
+token = params['_target']['token']
```

Comment:
I think for now the tasks will have to. For tasks that leverage RsAPI we'll be able to add some validation later. I intentionally left this out of this rev of the spec because I have questions about how we can enable without requiring a lot of copy-paste in metadata.


https://github.com/puppetlabs/puppet-specifications/pull/133

Reply:
The task example itself could error if token ended up being nil. But I'm ok ignoring it initially too.